packages/compass-e2e-tests/helpers/compass.ts

```diff
@@ -616,7 +620,67 @@ async function startCompass(opts: StartCompassOptions = {}): Promise<Compass> {
   debug('Starting compass via webdriverio with the following configuration:');
   debug(JSON.stringify(options, null, 2));
 
-  const browser = await remote(options);
+  let browser: CompassBrowser;
+
+  try {
+    browser = await remote(options);
+  } catch (err) {
+    debug('Failed to start remote webdriver session', {
+      error: (err as Error).stack,
+    });
+    // Sometimes when webdriver fails to start the remote session, we end up
+    // with a running app that hangs the test runner in CI causing the run to
+    // fail with idle timeout. We will try to clean up a potentially hanging app
+    // before rethrowing an error
+
+    // ps-list is ESM-only in recent versions.
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    const { default: psList }: typeof import('ps-list') = await eval(
+      `import('ps-list')`
+    );
+    const processList = await psList();
+
+    const filteredProcesses = processList.filter((p) => {
+      return (
+        p.ppid === process.pid &&
+        (p.cmd?.startsWith(binary) ||
+          /(MongoDB Compass|Electron|electron)/.test(p.name))
```

Comment:
Maybe an overkill, but cmd is marked as optional in package types, so I'm a bit afraid to only rely on it for the filtering